overlay.d/05core/usr/lib/dracut/modules.d/30ignition-coreos/coreos-boot-edit.sh

```diff
@@ -14,9 +14,22 @@ mount -o rw ${bootdev} ${bootmnt}
 # installed system (most likely by using `coreos-installer install --copy-network`).
 firstboot_network_dir_basename="coreos-firstboot-network"
 initramfs_firstboot_network_dir="${bootmnt}/${firstboot_network_dir_basename}"
-copy_firstboot_network_stamp="/run/coreos-copy-firstboot-network.stamp"
-if [ -f ${copy_firstboot_network_stamp} ]; then
-    rm -vrf ${initramfs_firstboot_network_dir}
-else
-    echo "info: no firstboot networking config files to clean from /boot. skipping"
+rm -vrf ${initramfs_firstboot_network_dir}
+
+# If root is specified, assume rootmap is already configured. Otherwise,
+# append rootmap kargs to the BLS configs.
+cmdline=( $(</proc/cmdline) )
+karg() {
+    local name="$1"
```

Comment:
Try this:

```suggestion
    local name="$1" value="${2:}"
```

That should fix the CI failure.

Reply:
Also, WDYT about putting that function at the top of the script to separate it out from the rest of the logic?

Reply:
The reason this is failing here is because we're running with `set -u` whereas the code in coreos-installer we copied this from just has `set -e` (though we should change it to use `set -euo pipefail` like here).

Reply:
+1 to putting it on top. Actually I think CI would still fail because I did `set -u`.

Reply:
It should be OK with `set -u` as long as you provide a default value on variables which may not be defined. This is what `value="${2:}"` does. That *should* be all that's needed.

Reply:
oh sorry, I misread. Right, `value="{2:}"` would work (didn't notice the `:` earlier).

Reply:
ah, I think the correct syntax should be `"${2:-}"`

Reply:
Yes sorry, you're right, that should've been `"${2:-}"`.